### PR TITLE
Decoupling PublishViewController

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details
In this PR we're properly encapsulating `PublishViewController`, and moving its View(s) to a standalone **nib**, setup with Autolayout.

@aerych May I trouble you yeeeet again?
Thanks in advance!!!

### Test
1. Add a new note
2. Press over the "Share Action" button (top right corner, third button)
3. Click over `Publish to the Web`

- [x] Verify the note is not published
- [ ] Verify that clicking `Publish` displays the `Publishing...` UI
- [ ] Verify that after a while (can be few seconds) the UI switches to the `Published` state
- [ ] Verify that clicking `Unpublish` causes the UI to display `Unpublishing...`
- [ ] Verify that shortly after, the popver is back to the Unpublished state
- [ ] Verify Dark / Light mode please!

**Note:** Lots and lots of improvement areas, just focusing (for now) on offloading the main (massive) nib!
### Release
These changes do not require release notes.

### CODE REVIEW

File: Simplenote/PublishViewController.swift [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        startListeningToNotifications()
+        applyStyle()
+    }
+
+    ///
+    ///
```
Comment (aerych) [inline Simplenote/PublishViewController.swift:70]: Was this going to be documentation?

File: Simplenote/NoteEditorViewController.m [hunk 2]
```diff
@@ -1091,7 +1079,9 @@ - (void)applyStyle
 
 - (void)showPublishPopover
 {
-    [self showViewController:self.publishViewController relativeToView:shareButton preferredEdge:NSMaxYEdge];
+    PublishViewController *viewController = [PublishViewController new];
+    [self showViewController:viewController relativeToView:shareButton preferredEdge:NSMaxYEdge];
+    self.publishViewController = viewController;
```
Comment (aerych) [inline Simplenote/NoteEditorViewController.m:1084]: Betcha you're gonna wanna do something like `self.publishViewController.delegate = self` right here yo. ;)

File: Simplenote/PublishViewController.swift [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    /// Returns the Publish Button's Internal State
+    ///
+    @objc
+    var publishButtonState: NSControl.StateValue {
+        publishButton.state
+    }
+
+    /// Old School delegate
+    ///
+    weak var delegate: PublishViewControllerDelegate?
```
Comment (aerych) [inline Simplenote/PublishViewController.swift:54]: This guy is probably gonna want a `@objc` so `NoteEditorViewController` can see it ;)

**Comment by jleandroperez** [comment]
@aerych Apologies!! Had those changes locally (but went unpushed!!). Ready for another peek!

**Thank you!!!**

**Comment by jleandroperez** [comment]
Thanks for reviewing this one sir!!

**Pull request merged**